src/main/java/com/nike/cerberus/util/AwsIamRoleArnParser.java

```diff
@@ -19,47 +19,76 @@
 
 import com.nike.backstopper.exception.ApiException;
 import com.nike.cerberus.error.InvalidIamRoleArnApiError;
+import org.apache.commons.lang3.StringUtils;
 
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 /**
  * Utility class for concatenating and parsing AWS IAM role ARNs.
  */
-// TODO: remove
 public class AwsIamRoleArnParser {
 
     public static final String AWS_IAM_ROLE_ARN_TEMPLATE = "arn:aws:iam::%s:role/%s";
 
     public static final String AWS_IAM_ROLE_ARN_REGEX = "^arn:aws:iam::(?<accountId>\\d+?):role/(?<roleName>.+)$";
 
-    public static final String AWS_IAM_PRINCIPAL_ARN_REGEX = "^arn:aws:(iam|sts)::.+$";
+    public static final String AWS_IAM_PRINCIPAL_ARN_REGEX = "^arn:aws:(iam|sts)::(?<accountId>\\d+?):(?!group).+?/(?<roleName>.+)$";
```

Comment:
This might be a nitpick but it seems like all of the regex in this class should probably be private.